taskt/Core/Automation/User32/GlobalHook.cs

```diff
@@ -0,0 +1,639 @@
+
+                    };
+
+                    User32Functions.SetWindowPosition(hwnd, 0, 0);
+
+                    GeneratedCommands.Add(moveWindowCommand);
+                }
+
+                //if tracking window sizes is set
+                if (_trackActivatedWindowSizes)
+                {
+                    //create rectangle from hwnd
+                    User32Functions.GetWindowRect(hwnd, out Rect windowRect);
+
+                    //do math to get height, etc
+                    var width = windowRect.right - windowRect.left;
+                    var height = windowRect.bottom - windowRect.top;
+
+                    //generate command to set window position
+                    ResizeWindowCommand reszWindowCommand = new ResizeWindowCommand
+                    {
+                        v_WindowName = windowName,
+                        v_XWindowSize = width.ToString(),
+                        v_YWindowSize = height.ToString(),
+                        v_Comment = "Generated by Screen Recorder @ " + DateTime.Now.ToString()
+
+                    };
+
+                    //add to list
+                    GeneratedCommands.Add(reszWindowCommand);
+                }
+            }
+        }
+        //build pause command
+        private static void BuildPauseCommand()
+        {
+
+            if (_stopWatch.ElapsedMilliseconds < 1)
+            {
+                return;
+            }
+
+            _stopWatch.Stop();
+            var pauseTime = _stopWatch.ElapsedMilliseconds;
+            var pauseCommand = new PauseScriptCommand
+            {
+                v_PauseLength = pauseTime.ToString()
+            };
+            GeneratedCommands.Add(pauseCommand);
+            _stopWatch.Restart();
+        }
+        private static IntPtr SetKeyboardHook(LowLevelKeyboardProc proc)
+
+        {
+
+            using (Process curProcess = Process.GetCurrentProcess())
+
+            using (ProcessModule curModule = curProcess.MainModule)
+
+            {
+
+                return SetWindowsHookEx(WhKeyboardLl, proc,
+
+                    GetModuleHandle(curModule.ModuleName), 0);
+
+            }
+
+        }
+        private static IntPtr SetMouseHook(LowLevelMouseProc proc)
+
+        {
+
+            using (Process curProcess = Process.GetCurrentProcess())
+
+            using (ProcessModule curModule = curProcess.MainModule)
+
+            {
+
+                return SetWindowsHookEx(WhMouseLl, proc, GetModuleHandle(curModule.ModuleName), 0);
+
+            }
+
+        }
+
+        private static IntPtr _winEventHook;
+        private static SystemEventHandler _winEventHookHandler;
+        private static StringBuilder _buffer = new StringBuilder(512);
+
+        #region User32 Keyboard Mouse
+        private delegate IntPtr LowLevelKeyboardProc(int nCode, IntPtr wParam, IntPtr lParam);
+        private delegate IntPtr LowLevelMouseProc(int nCode, IntPtr wParam, IntPtr lParam);
```

Comment:
Can this file be broken up into different classes by #region? Otherwise maybe move these properties to the top of the file. It's a bit messy from here down.

Reply:
Moved the declarations/imports to the top of the file.